pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Schema.java

```diff
@@ -120,6 +123,36 @@ default T decode(byte[] bytes, byte[] schemaVersion) {
         return decode(bytes);
     }
 
+    /**
+     * Decode a byte array into an object using the schema definition and deserializer implementation.
+     *
+     * @param input the inputstream to decode
+     * @return the deserialized object
+     */
+    default T decode(InputStream input) {
+        try {
+            return decode(IOUtils.readAllBytes(input));
```

Comment:
```suggestion
            return decode(org.apache.commons.io.IOUtils.toByteArray(input));
```

Reply:
There is no apache-commons-io dependency in client-api, so it can't be work. same as below